app/frontend/actions.ts

```diff
@@ -1255,89 +1255,75 @@ export default ({setState, getState}: {setState: SetStateFn; getState: GetStateF
     })
   }
 
-  /* Pool Owner */
-
-  const deserializeTransactionFile = (file) => {
-    if (!file || !file.cborHex) {
-      throw NamedError('PoolRegInvalidFileFormat')
-    }
-
-    const unsignedTxDecoded = decode(file.cborHex)
-    const parsedTx = parseUnsignedTx(unsignedTxDecoded)
-    return parsedTx
-  }
+  /* POOL OWNER */
 
-  const loadPoolCertificateTx = async (state: State, fileObj) => {
+  const loadPoolCertificateTx = async (state: State, file: string) => {
```

Comment:
I'd prefer to extract the logic of parsing and translating that to a txPlan to a separate utility function which would have a clear interface and no side-effect on the state. That way we can also unit-test it

Reply:
I separated the logic of parsing the file into a helper function, and converting it to txPlan to another, it seemed strange to me for a function to both parse JSON and also receive a staking address as an argument. Is there any problem with this approach? As I mentioned in the PR description, I didn't get to refactor everything since we don't want to delay release much longer.

Reply:
I see, I checked the refactor you did and looks good to me